js/damage.js

```diff
@@ -305,7 +308,10 @@ function getDamageResult(attacker, defender, move, field) {
         description.weather = field.weather;
     } else if (gen >= 6 && move.name === "Knock Off" && !(defender.item === "" ||
             (defender.name === "Giratina-O" && defender.item === "Griseous Orb") ||
-            (defender.name.indexOf("Arceus") !== -1 && defender.item.indexOf("Plate") !== -1))) {
+            (defender.name.indexOf("Arceus") !== -1 && defender.item.indexOf("Plate") !== -1) ||
+            (defender.name.indexOf("Genesect") !== -1 && defender.item.indexOf("Drive") !== -1) ||
+            (defender.name.indexOf("Primal Groudon") && defender.item === "Red Orb") ||
+            (defender.name.indexOf("Primal Kyogre") && defender.item === "Blue Orb"))) {
```

Comment:
Just checking for `"Groudon"` and `"Kyogre"` here is fine (like the ones above), since this stuff applies to all formes of these Pokemon.